Fix editorconfig

lbcjbb opened the pull request:
- Add missing .editorconfig
- Use process.env.INIT_CWD defined by npm to determine the root path rather than to use the git system command.

Commit:
Override root .gitignore file for not ignore .editorconfig in packages/lbc-editorconfig

Commit:
Use INIT_CWD defined by npm to find the root path of the project

The root project may not be a git repository.
By exemple when the root project is built in a Docker container in
several stages to use the Docker cache to cache the installation of
all Node.js dependencies in node_modules.
The first stage is used to copy only needed files (package*.json, npm
scripts, etc...) before to run the command `npm install`.
During the second stage, we copy the whole project inside the
container and run it.

Rather to use the git system command to determine the root path of the
project, we use the INIT_CWD defined by npm.

pa-bru commented on packages/editorconfig/index.js:
```diff
@@ -3,8 +3,7 @@ const fs = require('fs')
 const path = require('path')
 const execSync = require('child_process').execSync
```
you can remove this useless line

lbcjbb replied:
Done

pa-bru commented on packages/editorconfig/index.js:
```diff
@@ -3,8 +3,7 @@ const fs = require('fs')
 const source = path.join(__dirname, '.editorconfig')
-const gitRoot = execSync('git rev-parse --show-toplevel').toString().trim()
-const dest = path.join(gitRoot, '.editorconfig')
+const dest = path.join(process.env.INIT_CWD, '.editorconfig')
```
are you sure it resolves the path you really want ? this env variable gives you the full path you were in when you ran npm run. it doesn't necessarily give you the top level directory path.

may you want to add the editorconfig file elsewhere than in the root directory ?

pa-bru replied:
yes

Commit:
Revert use of INIT_CWD to determine the root path

npm can be launched in another directory than the root path of the
project. We rather use the value of an custom environment variable
APP_ROOT_PATH defined by the user to avoid call to git.
This variable is already used by other node modules like
app-root-path.

Commit:
Publish

 - @leboncoin/editorconfig@1.0.13
 - @leboncoin/gitignore@1.0.9
 - @leboncoin/internal-dev@2.0.22
 - @leboncoin/npmrc@1.0.7
 - @leboncoin/prettier@1.0.4
 - @leboncoin/root-path@0.0.1
